### PR TITLE
Admin add coupon

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/75151

## WHAT (optional)
- Add feature admin add coupon

## HOW
- Add feature admin add coupon

## Evidence (Screenshot or Video)
Screenshot from 2024-05-10 15-23-24
Screenshot from 2024-05-10 15-23-10
Screenshot from 2024-05-10 15-10-44
Screenshot from 2024-05-10 15-09-02

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/dto/coupon/createCoupon.dto.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  @Validate(IsPercentage)
+  percentage: number;
+
+  @Validate(IsBeforeEndDate)
+  @Validate(IsAfterCurrentDate)
+  startDate: Date;
+
+  @Validate(IsAfterStartDate)
+  @Validate(IsAfterCurrentDate)
+  endDate: Date;
```
Comment (huongnt-2545) [inline src/dto/coupon/createCoupon.dto.ts]: em bổ sung validate required cho cả 4 fields này nhé

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**